daisi/src/cpps/model/task_event.h

```diff
@@ -0,0 +1,36 @@
+// Copyright 2023 The SOLA authors
+//
+// This file is part of DAISI.
+//
+// DAISI is free software: you can redistribute it and/or modify it under the terms of the GNU
+// General Public License as published by the Free Software Foundation; version 2.
+//
+// DAISI is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+// the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+// Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along with DAISI. If not, see
+// <https://www.gnu.org/licenses/>.
+//
+// SPDX-License-Identifier: GPL-2.0-only
+
+#ifndef DAISI_MODEL_TASK_EVENT_H_
+#define DAISI_MODEL_TASK_EVENT_H_
+
+namespace daisi::cpps {
+
+enum class TaskEvent {
+  kCreated = 0,
+  kQueued = 1,
+  kStarted = 2,
+  kArrived = 3,
+  kLoaded = 4,
+  kUnloaded = 5,
+  kFinished = 6,
+  kError = 12,
```

Comment:
Why the gap in numbers? In that case kError could also start at 100.

Reply:
I guess, its relates to some legacy states which have been just removed

Reply:
In OrderStates kError was also 12 so I kept the number. It also works as a buffer in case TaskEvent needs more enums.

Reply:
reverted to using OrderStates